analyzer/javacg-opal/src/main/java/eu/fasten/analyzer/javacgopal/data/callgraph/ExtendedRevisionCallGraph.java

```diff
@@ -34,214 +42,484 @@
      */
-    public void clear(final Boolean excessiveRemove) {
-        if (excessiveRemove) {
-            this.graph.parallelStream().forEach(i -> {
-                i[0] = null;
-                i[1] = null;
-            });
+    public ExtendedRevisionCallGraph(final JSONObject json) throws JSONException {
+
+        super(json.getString("forge"),
+            json.getString("product"),
+            json.getString("version"),
+            getTimeStamp(json),
+            Dependency.depset(json.getJSONArray("depset")),
+            new ArrayList<>());
 
-            this.classHierarchy.forEach((fastenURI, type) -> {
-                fastenURI = null;
-                type.methods.parallelStream().forEach(i -> i = null);
-                type.superClasses.parallelStream().forEach(i -> i = null);
-                type.superInterfaces.parallelStream().forEach(i -> i = null);
-            });
-            this.classHierarchy.clear();
-        } else {
-            this.graph = null;
-            this.classHierarchy = null;
+        this.cgGenerator = json.getString("generator");
+        this.graph = new Graph(json.getJSONObject("graph"));
+        this.classHierarchy = classHierarchy(json.getJSONObject("cha"));
+
+    }
+
+    /** If timestamp is present in the JSON set it otherwise set it to -1. */
+    private static long getTimeStamp(JSONObject json) {
+        try {
+            return json.getLong("timestamp");
+        } catch (final JSONException exception) {
+            logger.warn("No timestamp provided: assuming -1");
+            return -1;
         }
     }
 
-    public void clear() {
-        clear(false);
+    /**
+     * Creates {@link ExtendedRevisionCallGraph} using OPAL call graph generator for a given maven
+     * coordinate. It also sets the forge to "mvn".
+     * @param coordinate maven coordinate of the revision to be processed.
+     * @param timestamp  timestamp of the revision release.
+     * @return {@link ExtendedRevisionCallGraph} of the given coordinate.
+     * @throws FileNotFoundException in case there is no jar file for the given coordinate on the
+     *                               Maven central it throws this exception.
+     */
+    public static ExtendedRevisionCallGraph create(final MavenCoordinate coordinate,
+                                                   final long timestamp)
+        throws FileNotFoundException {
+        final var partialCallGraph = new PartialCallGraph(
+            MavenCoordinate.MavenResolver.downloadJar(coordinate.getCoordinate())
+                .orElseThrow(RuntimeException::new)
+        );
+
+        return new ExtendedRevisionCallGraph("mvn", coordinate.getProduct(),
+            coordinate.getVersionConstraint(), timestamp, partialCallGraph.getGENERATOR(),
+            MavenCoordinate.MavenResolver.resolveDependencies(coordinate.getCoordinate()),
+            partialCallGraph.getClassHierarchy(),
+            partialCallGraph.getGraph());
     }
```

Comment:
I like the new ExtendedRevisionCallGraph, but I have some concerns about this method. I don't think we should have a dependency from core to any analyser [in case this class is going to be moved to the core module]. This method depends on MavenCoordinate and PartialCallGraph from OPAL module. I think we should rather have a method in PartialCallGraph of each analyser that converts it to ExtendedRCG.

Reply:
Thanks for the review, I moved it to the PartialCallGraph. Yes I agree, actually, in the beginning, it was there I moved it for cosmetic reasons. But anyway you are right it's better not to make core dependent on the analyzers.